nni/algorithms/compression/v2/pytorch/base/compressor.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import collections
+import logging
+from typing import List, Dict, Optional, OrderedDict, Tuple, Any
+
+import torch
+from torch.nn import Module
+from torch.tensor import Tensor
+
+from nni.common.graph_utils import TorchModuleGraph
+
+_logger = logging.getLogger(__name__)
+
+__all__ = ['LayerInfo', 'Compressor']
+
+
+class LayerInfo:
+    def __init__(self, name: str, module: Module):
+        self.module = module
+        self.name = name
+        self.type = type(module).__name__
+
+
+def _setattr(model: Module, name: str, module: Module):
```

Comment:
looks like duplicated with `get_module_by_name` , please use this function instead which takes some corner case into account.

https://github.com/microsoft/nni/blob/d8127e02841f7f2afb923c22c91b254155544db8/nni/compression/pytorch/utils/utils.py#L9

Reply:
thanks, it is better, I have modified it.